### ISSUE TITLE
Question regarding datapower object count metrics

### ISSUE THREAD

**Issue by djarcan** [IBM/apiconnect-trawler#62]
Hi,

thank you for providing the opportunity to adjust the timeout in the datapower_net.
We noticed in our logs that something is gathered now:

```
2023-03-16T08:05:59.902914 [trawl:INFO] Trawling for metrics...
2023-03-16T08:06:00.278873 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:00.443304 [datap:INFO] DataPowers in list: 6
2023-03-16T08:06:00.677276 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:00.825884 [datap:INFO] DataPowers in list: 6
2023-03-16T08:06:01.091210 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:01.275606 [datap:INFO] DataPowers in list: 6
2023-03-16T08:06:01.547945 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:01.720889 [datap:INFO] DataPowers in list: 6
2023-03-16T08:06:01.973523 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:02.140620 [datap:INFO] DataPowers in list: 6
2023-03-16T08:06:02.369868 [datap:INFO] Processing status provider ObjectInstanceCounts
2023-03-16T08:06:02.523395 [datap:INFO] DataPowers in list: 6
```
Unfortunately however, data are not showing up in Grafana yet.
We noticed that instead of querying the `ObjectStatus`, the `ObjectInstanceCounts` is fetched now.
**Q1: Does this change the identifier where we would find them?**

Example identifieres we would expect to find:
- datapower_GatewayPeeringManager_total
- datapower_GatewayPeering_total
- datapower_APIGateway_total
- datapower_APICollection_total
- datapower_APIPath_total

**Comment by rickymoorhouse** [comment]
No this won't change the identifier and I have confirmed I'm seeing that list of metrics you mentioned on my test system exactly as they were before.

The previous call was pulling back the lists of objects and counting them in trawler where as this returns the same object types but jsut the count so we can pass it straight into the metrics.

**Comment by djarcan** [comment]
Thanks, that's good to know.

I activated debug logs and received:

```
2023-03-16T10:31:35.274148 [datap:INFO] Processing status provider ObjectInstanceCounts                                                                                            
2023-03-16T10:31:35.379187 [datap:DBUG] {"error": ["Invalid class name in the URI."]} 
```

Maybe you have a hint already regarding that. I will check for further information.

**Comment by rickymoorhouse** [comment]
Yes this was only added in 10.5+ - do you have plans to upgrade to 10.5?

**Comment by rickymoorhouse** [comment]
Will have a think about how to fall back to the old method in earlier versions

**Comment by rickymoorhouse** [comment]
Ok I think I have a fix that will use ObjectStatus for DP below 10.5.0.0 and ObjectInstanceCount for newer datapowers

**Comment by djarcan** [comment]
It's on our roadmap to upgrade to API Connect 10.0.5 and it's corresponding GW version which I assume to be 10.5 from your statements. However, I cannot say when this will be happening.

Please create a new image tag after the PR is merged (since we always use pinned versions and not latest)
Thank you very much for the fast fix - It's very much appreciated ❤️

**Comment by rickymoorhouse** [comment]
This is in the new image tagged v1.7.1 - took a little longer to validate as most of our environments were on 10.0.5 variants

**Comment by djarcan** [comment]
It seems that the timeout was not properly propagated from the DataPowerNet to DataPower class. See: https://github.com/IBM/apiconnect-trawler/pull/64

**Comment by rickymoorhouse** [comment]
Thanks for the fix - this will be in the release I just published v1.7.2

**Comment by rickymoorhouse** [comment]
Fantastic news - and thanks for your help 👍
